source/views/sis/course-search/index.js

```diff
@@ -0,0 +1,214 @@
+// @flow
+
+import * as React from 'react'
+import {StyleSheet, ScrollView, View, Animated, Dimensions} from 'react-native'
+import {TabBarIcon} from '../../components/tabbar-icon'
+import * as c from '../../components/colors'
+import {CourseSearchBar} from '../components/searchbar'
+import debounce from 'lodash/debounce'
+import {updateCourseData} from '../../../flux/parts/sis'
+import type {CourseType} from '../../../lib/course-search'
+import type {ReduxState} from '../../../flux'
+import type {TopLevelViewPropsType} from '../../types'
+import {connect} from 'react-redux'
+import groupBy from 'lodash/groupBy'
+import sortBy from 'lodash/sortBy'
+import toPairs from 'lodash/toPairs'
+import {CourseSearchResultsList} from './list'
+
+type ReactProps = TopLevelViewPropsType
+
+type ReduxStateProps = {
+	allCourses: Array<CourseType>,
+	courseDataState: string,
+}
+
+type ReduxDispatchProps = {
+	updateCourseData: () => any,
+}
+
+type Props = ReactProps &
+	ReduxStateProps &
+	ReduxDispatchProps & {
+		navigation: {state: {params: {}}},
+	}
+
+type State = {
+	searchResults: Array<{title: string, data: Array<CourseType>}>,
+	searchActive: boolean,
+}
+
+class CourseSearchView extends React.PureComponent<Props, State> {
+	static navigationOptions = {
+		tabBarLabel: 'Course Search',
+		tabBarIcon: TabBarIcon('search'),
+		title: 'SIS',
+	}
+
+	state = {
+		searchResults: [],
+		searchActive: false,
+	}
+
+	componentDidMount() {
+		this.props.updateCourseData()
+	}
+
+	searchBar: any = null
+	headerOpacity = new Animated.Value(1)
+	searchBarTop = new Animated.Value(71)
+	containerHeight = new Animated.Value(125)
+
+	_performSearch = (text: string | Object) => {
+		// Android clear button returns an object
+		// if (typeof text !== 'string') {
+		// 	return this.props.onSearch(null)
+		// }
+		const query = text.toLowerCase()
+
+		let results = this.props.allCourses.filter(course => {
+			const instructors = course.instructors
+				? course.instructors.toString().toLowerCase()
+				: []
+			return (
```

Comment:
Minor request: I'd love to be able to search by deptnum too. Maybe `|| \`${course.departments.join('/')} ${course.number.toString()}${course.section || ''}\`.startsWith(query)`?

Reply:
(resolved)